AEPCampaignClassic/Sources/CampaignClassic+PublicAPI.swift

```diff
@@ -16,4 +16,89 @@ import Foundation
+
+        var eventData = [CampaignClassicConstants.EventDataKeys.CampaignClassic.REGISTER_DEVICE: true,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_TOKEN: token.hexDescription,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.USER_KEY: userKey,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO: deviceInfo] as [String: Any]
+
+        // attach additional parameters only if they are available
+        if let additionalParameter = additionalParameter {
+            eventData[CampaignClassicConstants.EventDataKeys.CampaignClassic.ADDITIONAL_PARAMETERS] = additionalParameter
+        }
+
+        let event = Event(name: "CampaignClassic Register Device",
+                          type: CampaignClassicConstants.SDKEventType.CAMPAIGN_CLASSIC,
+                          source: EventSource.requestContent,
+                          data: eventData)
+
+        MobileCore.dispatch(event: event, responseCallback: { responseEvent in
+            guard let responseEvent = responseEvent else {
+                callback(false)
+                return
+            }
+
+            guard let registrationResult = responseEvent.data?[CampaignClassicConstants.EventDataKeys.CampaignClassic.REGISTRATION_STATUS] as? Bool else {
+                callback(false)
+                return
+            }
+
+            callback(registrationResult)
+
+        })
+    }
+
+    /// Sends tracking information to the configured Adobe Campaign Classic server.
+    /// Use this API to send tracking info on receiving a notification (silent push).
+    ///
+    /// - Parameter trackingInfo :  a dictionary containing `_dId` and `_mId` received in the push message payload, or in the
+    ///                             launch options before opening the application
+    static func trackNotificationClick(trackingInfo: [String: String]) {
+        let eventData = [CampaignClassicConstants.EventDataKeys.CampaignClassic.TRACK_CLICK: true,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.TRACK_INFO: trackingInfo] as [String: Any]
+
+        let event = Event(name: "CampaignClassic Track Notification Receive",
+                          type: CampaignClassicConstants.SDKEventType.CAMPAIGN_CLASSIC,
+                          source: EventSource.requestContent,
+                          data: eventData)
+        MobileCore.dispatch(event: event)
+    }
+
+    /// Sends tracking information to the configured Adobe Campaign Classic server.
+    /// Use this API to send tracking info when the application is opened following a notification.
+    ///
+    /// - Parameter trackingInfo :  a dictionary containing `_dId` and `_mId` received in the push message payload, or in the
+    ///                             launch options before opening the application
+    static func trackNotificationReceive(trackingInfo: [String: String]) {
```

Comment:
Similar feedback here as for notification click, suggest updating the API to
`static func trackNotificationReceive(withData data: [String: Any])`